packages/flutter/lib/src/material/expansion_tile.dart

```diff
@@ -80,6 +84,22 @@ class ExpansionTile extends StatefulWidget {
   /// Specifies if the list tile is initially expanded (true) or collapsed (false, the default).
   final bool initiallyExpanded;
 
+  /// The color of the header when the tile is collapsed.
```

Comment:
It might actually be better to simply pass in a [ColorTween](https://master-api.flutter.dev/flutter/animation/ColorTween-class.html). For example:
```dart
/// Docs
final ColorTween headerColorTween;

/// Docs
final ColorTween iconColorTween;
```

This makes the API simpler since there are only two parameters instead of four.

Reply:
This might require moving some refactoring of _ExpansionTileState._headerColorTween and _ExpansionTileState._iconColorTween, since they're currently final fields.

Reply:
@shihaohong, pass a `ColorTween` doesn't seem to be the default implementation on other widgets, e.g, `BottomNavigationBar` takes `unselectedItemColor` and `selectedItemColor`.

What I noticed is that they use more semantic names, which we could apply here. What do you think? I've thought about the following names (same I used on the tests):

- expandedTitleColor
- expandedIconColor
- collapsedTitleColor
- collapsedIconColor

Reply:
The internals of the widget still end up using a `ColorTween`, so I feel like it does make the API cleaner without forcing anything new onto the user. We could add some API documentation that they could use the `ColorTween` to simply toggle a color if they wanted to as well. 

A key difference would be that we would be replacing the entire `ColorTween` instead of updating values on a single existing `ColorTween`. There might be implications to that that I do not understand. @HansMuller do you know if this might cause any problems or observable differences in behavior?

Reply:
Found a section of the [Tween API docs](https://master-api.flutter.dev/flutter/animation/Tween-class.html) on mutable Tweens. Making this type of change could have a performance implication

Reply:
I see. Should we wait for @HansMuller's input or can I proceed with the renaming properties approach?

Reply:
Do you mind testing to see if there are any observable differences between the two? Just building both out on a device and seeing if performance is impacted, does changing the tween’s colors do anything different, etc

Reply:
I'll try to do that in the following days.

Reply:
Thank you! Let me know if you need any help or suggestions

Reply:
I've updated the PR with your suggested changes and test them using this stateful version of the [expansion_tile_sample.dart example](https://gist.github.com/arthurdenner/1780d3594a06eb06b94474d45c23301a).

Performance doesn't seem to be affected, but the icon is not updating when calling `setState`. If I copy the logic from `_ExpansionTileState.didChangeDependencies` to `_ExpansionTileState.didUpdateWidget`, they do, but I'm not sure about this change, so I didn't commit it.

If I was clear on the explanation, what are your thoughts on this? I can commit these changes so that you can test too if you prefer and remove them before the merge.

Reply:
This makes sense. `didChangeDependencies` is called immediately after `initState`, and in the original implementation, a private `ColorTween` is created in `initState` and its begin and end values were set to the passed in `ColorTween`s begin and end values.

On the other hand, `didUpdateWidget` is called after every `build`, which is necessary here if you ever update ColorTween being used using `setState`.

Reply:
It's pretty dense, but the [State](https://api.flutter.dev/flutter/widgets/State-class.html) documentation pretty much outlines the lifecycle for State objects.

Reply:
Nice, thanks! I'll read it. Should I commit and push these changes then?

Reply:
Go ahead and commit it so I (and whoever else that reads this PR) can take a look and see if it works out :)

Reply:
Done. I've updated the `expansion_tile_sample` example too for an easier demo.